### PR TITLE
SPOI-10346-kafka-to-hdfs-transform : Apps, configs, README.

### PR DESCRIPTION
**Pull request by yogidevendra**

### CODE REVIEW

**Comment by yogidevendra** [comment]
@chaithu14 Could you please review this?

**Comment by yogidevendra** [comment]
@shubham-pathak22  Could you please review this?

File: kafka-to-hdfs-filter-transform/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+### Description
+This Kafka to HDFS Filter with Transform Application continuously reads json string messages from configured kafka topic(s), filters based on the filter criteria, transforms specified fields and writes each message as a json string in HDFS file(s). This application derives schema from input json.
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: This -> the. Also you could add a title "Kafka to HDFS with Filter and Transform Application" and name this section as Introduction

Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: Transforms specified fields into ?

Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: "Derives schema from input json" is kind of confusing.  You may omit this line.

File: kafka-to-hdfs-filter-transform/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Add custom logic to the application template and launch. Please watch the [walkthrough video](https://www.youtube.com/watch?v=S6CB4XgRHCE) to add custom logic to the application template.
+
+<iframe src="https://www.youtube.com/embed/S6CB4XgRHCE?enablejsapi=1" allowfullscreen="allowfullscreen" class="video" id="advancedVideo" ga-track="advancedVideo"></iframe>
+
+### Logical Plan
+
+Here is a preview of the logical plan of the application template
+
+![Logical Plan](https://www.datatorrent.com/wp-content/uploads/2016/12/kafka_to_hdfs_filter_DAG.png)
+
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: The image has operator named CSV Parser while the app uses a JsonParser. Also i don't see transform operator either. Please update the image

File: kafka-to-hdfs-filter-transform/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+To customize this application, user can configure it to run it with different schema at the launch time. 
+- The application scales linearly with the number of Kafka brokers and Kafka topics.
+- The application is fault tolerant and can withstand node and cluster outages without data loss.
+- The application is also highly performant and can process faster than Kafka broker can produce per topic.
+- It is extremely easy to add custom logic to get your business value without worrying about connectivity and operational details of Kafka consumer and HDFS writer.
+- The only configuration user needs to provide is source Kafka broker list, topic list, input schema, filter criteria and destination HDFS path, filename.
+- This enterprise grade application template will dramatically reduce your time to market and cost of operations.
+
+Import the application from DataTorrent AppHub and launch it to read your data from Kafka topics, filters based on the filter criteria, transforms specified fields and write to HDFS. Follow the tutorial videos or walkthrough document below to launch the template and add custom logic to process the data during ingestion.
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: "it to read your data from Kafka topics, filters based on the filter criteria, transforms specified fields and write to HDFS." this is repetitive. You can just say, import the app, and follow the tutorial to configure, customize and launch the application

File: kafka-to-hdfs-filter-transform/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+### Jumpstart
+Import and run application template as an operable proof of concept. Please watch the [walkthrough video](https://www.youtube.com/watch?v=taHc_QJUfBg) to import and launch the application.
+
+<iframe src="https://www.youtube.com/embed/taHc_QJUfBg?enablejsapi=1" allowfullscreen="allowfullscreen" class="video" id="basicVideo" ga-track="basicVideo"></iframe>
+
+### Productize
+Add custom logic to the application template and launch. Please watch the [walkthrough video](https://www.youtube.com/watch?v=S6CB4XgRHCE) to add custom logic to the application template.
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: As the video is embedded, I dont see the need to add the link to "walkthrough video" . You may choose to ignore this comment if you find it appropriate

File: kafka-to-hdfs-filter-transform/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Here is a preview of the logical plan of the application template
+
+![Logical Plan](https://www.datatorrent.com/wp-content/uploads/2016/12/kafka_to_hdfs_filter_DAG.png)
+
+### Launch App Properties
+
+Here is a preview of the properties to be set at application launch
+
+![Launch App Properties](https://www.datatorrent.com/wp-content/uploads/2016/12/kafka_to_hdfs_filter_properties.png)
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/README.md]: Is this image of the new UI ?

File: kafka-to-hdfs-filter-transform/src/test/java/com/datatorrent/apps/PojoEvent.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  public int getAmount()
+  {
+    return amount;
+  }
+
+  public void setAmount(int amount)
+  {
+    this.amount = amount;
+  }
+}
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/src/test/java/com/datatorrent/apps/PojoEvent.java]: Add a new line

File: kafka-to-hdfs-filter-transform/pom.xml [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+  <!-- change these to the appropriate values -->
+  <name>Kafka to HDFS Filter Transform App</name>
+  <description>
+    Ingest and Transform filtered messages from kafka to hadoop HDFS.
+  </description>
+
+  <properties>
+    <!-- change this if you desire to use a different version of Apex Core -->
+    <apex.version>3.4.0</apex.version>
+    <malhar.version>3.7.0-SNAPSHOT</malhar.version>
```
Comment (shubham-pathak22) [inline kafka-to-hdfs-filter-transform/pom.xml]: Can the version be snapshot ? or should it be the latest release version ?

File: kafka-to-hdfs-filter-transform/README.md (hunk 1 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/README.md]: done

File: kafka-to-hdfs-filter-transform/README.md (hunk 3 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/README.md]: done.

File: kafka-to-hdfs-filter-transform/README.md (hunk 4 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/README.md]: This is part of standard template.

File: kafka-to-hdfs-filter-transform/README.md (hunk 2 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/README.md]: There will separate PR to change video links, images. This is just to get app code merged into master. 
Video, doc, images is not included in this PR.

File: kafka-to-hdfs-filter-transform/README.md (hunk 5 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/README.md]: Video, doc, images is not included in this PR.

File: kafka-to-hdfs-filter-transform/src/test/java/com/datatorrent/apps/PojoEvent.java (hunk 6 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/src/test/java/com/datatorrent/apps/PojoEvent.java]: done

File: kafka-to-hdfs-filter-transform/pom.xml (hunk 7 above)
Comment (yogidevendra) [inline kafka-to-hdfs-filter-transform/pom.xml]: SNAPSHOT to be used if it is dependent on some changes which are not available in the released version.

**Pull request merged**